chapters/iterative-search.qmd

```diff
@@ -662,4 +666,759 @@ We’ll encounter even more Bayesian methods in subsequent chapters. In @sec-cls
+
+:::
+
+:::
+
+::: {.column width="15%"}
+:::
+
+:::: 
+
+The complete search produced new optimal values at iterations `r bo_best_chr`. The decrease in RMSE shows few meaningful improvements among negligible decreases: `r rmse_best_chr`. The search process was able to effectively tune the SVM model and find very good parameters. 
+
+The cost of the Gaussian process fit does become more expensive as new candidates are sampled, making the process somewhat less efficient than the previous approaches. The average time to compute each candidate was `r prettyunits::pretty_sec((initial_time[3] + bo_time[3])/nrow(bo_mtr))`, although the cost of each increases with iterations. 
+
+## Example: Tuning a Neural Network {#sec-bayes-opt-nnet}
+
+We revisit the barley data, where the percentage of barley oil in a mixture is estimated using spectrographic measurements. Our previous work on these data  addressed the high correlation between predictors via feature engineering with principal components. 
+
+In this chapter, we’ll take a more sophisticated approach to preprocessing this type of data using signal processing techniques, specifically the **Savitzky-Golay** (SG) method [@Schafer2011;@rinnan2009review]. This technique incorporates two main goals: to smooth the data from location to location (i.e., across the x-axis in @fig-barley-data(b)) and to difference the data. Smoothing attempts to reduce measurement noise without diminishing the data patterns that could predict the outcome. Differencing can remove the correlation between locations and might also mitigate other issues, such as systematic background noise. 
+
+For smoothing, the SG procedure uses a moving window across a sample’s location and, within each window, uses a polynomial basis expansion to produce a smoother representation of the data. The window width and the polynomial degree are both tuning parameters. 
+
+Differencing has a single tuning parameter: the difference order. A zero-order difference leaves the data as-is (after smoothing). A first-order difference is the smoothed value minus the smoothed value for the next location (and so on). Savitzky and Golay's method uses the polynomial estimates to estimate the differences/derivatives, instead of manually subtracting columns after smoothing. 
+
+@fig-savitzky-golay shows an example of a small set of locations from a single spectra in @fig-barley-data. The top row shows the effect of smoothing when differencing is not used. For a linear smoother (i.e., a polynomial degree of one), the process can over-smooth the data such that the nonlinear portions of the raw data lose significant amounts of information. This issue goes away as the polynomial order is increased. However, the higher degree polynomial nearly interpolates between data points and may not be smoothing the data in any meaningful way. 
```

Comment:
The SG computations require a version of the measure package currently in PR form. It also depends on devel dials but that should go to CRAN shortly.